safe-svg.php

```diff
@@ -464,16 +464,43 @@ protected function svg_dimensions( $svg ) {
             $height = 0;
             if ( $svg ) {
                 $attributes = $svg->attributes();
-                if ( isset( $attributes->width, $attributes->height ) && is_numeric( (float)$attributes->width ) && is_numeric( (float)$attributes->height ) ) {
-                    $width  = floatval( $attributes->width );
-                    $height = floatval( $attributes->height );
-                } elseif ( isset( $attributes->viewBox ) ) {
+
+                if ( isset( $attributes->viewBox ) ) {
                     $sizes = explode( ' ', $attributes->viewBox );
                     if ( isset( $sizes[2], $sizes[3] ) ) {
-                        $width  = floatval( $sizes[2] );
-                        $height = floatval( $sizes[3] );
+                        $viewbox_width  = floatval( $sizes[2] );
+                        $viewbox_height = floatval( $sizes[3] );
                     }
+                }
+
+                if ( isset( $attributes->width, $attributes->height ) && is_numeric( (float) $attributes->width ) && is_numeric( (float) $attributes->height ) && ! $this->str_ends_with( (string) $attributes->width, '%' ) && ! $this->str_ends_with( (string) $attributes->height, '%' ) ) {
+                    $attr_width  = floatval( $attributes->width );
+                    $attr_height = floatval( $attributes->height );
+                }
+
+                /**
+                 * Decide which attributes of the SVG we use first for image tag dimensions.
+                 *
+                 * We default to using the parameters in the viewbox attribute but
+                 * that can be overridden using this filter if you'd prefer to use
+                 * the width and height attributes.
+                 *
+                 * @hook safe_svg_use_width_height_attributes
+                 *
+                 * @param bool $false If the width & height attributes should be used first. Default false.
+                 * @param string $svg The file path to the SVG.
+                 *
+                 * @return bool If we should use the width & height attributes first or not.
```

Comment:
```suggestion
                 * @param {bool} $false If the width & height attributes should be used first. Default false.
                 * @param {string} $svg The file path to the SVG.
                 *
                 * @return {bool} If we should use the width & height attributes first or not.
```

According to these docs, we use [JSDoc for hook docblocks](https://github.com/10up/actions-wordpress/blob/stable/hookdocs-workflow.md).